property_tests/arbitraries/candid/constructed/variant_arb/base.ts

```diff
@@ -0,0 +1,125 @@
+import fc from 'fast-check';
+import { CandidMeta } from '../../candid_arb';
+import { CandidType } from '../../candid_type_arb';
+import { UniqueIdentifierArb } from '../../../unique_identifier_arb';
+import { JsFunctionNameArb } from '../../../js_function_name_arb';
+import { Variant } from '.';
+
+type Field = [string, CandidMeta<CandidType>];
+
+function VariantFieldsArb(
+    candidTypeArb: fc.Arbitrary<CandidMeta<CandidType>>
+): fc.Arbitrary<Field[]> {
+    return fc.uniqueArray(fc.tuple(JsFunctionNameArb, candidTypeArb), {
+        selector: (entry) => entry[0],
+        minLength: 1
+        // Although no minLength is technically required (according to the
+        // spec), the DFX CLI itself currently errors out trying to pass
+        // an empty object.
```

Comment:
Is this still a problem? We should open an issue somewhere

Reply:
We should also open an issue for the tuple inconsistency we found